tests/conftest.py

```diff
@@ -18,15 +23,61 @@
 PROXY_PORT = 8080
 
 
+def _is_coroutine(obj):
+    """Check to see if an object is really a coroutine."""
+    return curio.meta.iscoroutinefunction(obj) or inspect.isgeneratorfunction(obj)
+
+
+@pytest.mark.tryfirst
+def pytest_pycollect_makeitem(collector, name, obj):
+    """A pytest hook to collect coroutines in a test module."""
+    if collector.funcnamefilter(name) and _is_coroutine(obj):
+        item = pytest.Function.from_parent(collector, name=name)
+        if "curio" in item.keywords:
+            return list(collector._genfunctions(name, obj))
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_pyfunc_call(pyfuncitem):
+    """Run curio marked test functions in a Curio kernel instead of a normal function call.
+    """
+    if pyfuncitem.get_closest_marker("curio"):
+        pyfuncitem.obj = wrap_in_sync(pyfuncitem.obj)
+    yield
+
+
+def wrap_in_sync(func):
+    """Return a sync wrapper around an async function executing it in a Kernel."""
+
+    @functools.wraps(func)
+    def inner(**kwargs):
+        coro = func(**kwargs)
+        curio.Kernel().run(coro, shutdown=True)
+
+    return inner
+
+
+# Fixture for explicitly running in Kernel instance.
+@pytest.fixture(scope="session")
+def kernel(request):
+    """Provide a Curio Kernel object for running co-routines."""
+    k = curio.Kernel(debug=[curio.debug.longblock, curio.debug.logcrash])
+    m = curio.monitor.Monitor(k)
+    request.addfinalizer(lambda: k.run(shutdown=True))
+    request.addfinalizer(m.close)
+    return k
```

Comment:
Is all of this mostly vendorized code from https://github.com/johnnoone/pytest-curio?

I see ^ is really tiny, so it not looking maintained anymore (5y+) isn't really a problem since the project is most likely "done" and doesn't even need maintenance. So I wouldn't be against adding it as a test dependency. If we do prefer to have this glue code as-is in our repo then _maybe_ at least let's put it somewhere isolated, eg in `tests/marks/curio.py`?

Reply:
There is a problem with this package. It's obsolete.
[This line](https://github.com/johnnoone/pytest-curio/blob/6c93257244c8bd9c0bcf5f717758fdf712a70ca3/pytest_curio/plugin.py#L16
) caused a pytest deprecation warning (which is an Exception during the build). So that I had to copy the plugin example from [there](https://github.com/dabeaz/curio/blob/ee27c4515759616747b9991857169924fdb52824/examples/pytest_plugin.py).

Reply:
So after that ^ I added the new commit a44909e70bde469157f21ec2430159e0c082df68 

If someone need to add new backend with custom pytest marks etc, they will need to do the same

Reply:
Okay, that all makes sense. I'd recommend we maybe just have a comment in our own `marks/curio` module explaining that it's a vendoring of `pytest-curio`, because that package is out of date.